### ISSUE TITLE
Flamegraph: Code preview

### ISSUE THREAD

**Issue by LouisInFlow** [pyroscope-io/pyroscope#20]
Would be cool if a preview of the relevant code can be shown from the flamegraph.

- Could have hovercard that shows function details
- Could link to line in github, or perhaps even open in their code editor

**Comment by Rperry2174** [comment]
Yeah this would be awesome... Right now we're trying to figure out a way to distinguish "our" code from library code. For example:

![image](https://user-images.githubusercontent.com/23323466/104824291-50e1fb00-5805-11eb-97e3-e6c4ea048dbb.png)

Here:
`services/customer/client.py:16` is part of our code while
`flask/app.py` is part of the flask package in python

So we could link to `services/customer/client.py:16` in our code and _if_ there was some way to know on the frontend that `flask/app.py` is not in our repo then we could just use that logic to determine whether or not to include the linking behavior.

Right now thats the blocker here (we may need to do more digging on the backend for this).

**Comment by Rperry2174** [comment]
Just ftr I think that we would probably try and do something like this:
![image](https://user-images.githubusercontent.com/23323466/104824435-83d8be80-5806-11eb-91ff-d2b017c1effc.png)

Which would keep the current behavior that "zooms" in on a node when you click on it. Then after it's clicked on we could make it slightly larger height and show a little github icon inside of there that would link to:

```
https://github.com/pyroscope-io/[REPO_NAME]/services/frontend/server.py:38
```

**Comment by LouisInFlow** [comment]
Easier way to distinguish on frontend may be just create a hard coded list. ```libraries-blacklist.js``` that looks like ```export new set(['list of top 1000 libraries for each language'])``` then check when rendering each box

**Comment by LouisInFlow** [comment]
Think it'll look ok to add the github logo to every single box? Also how would user access the link for the boxes that are too thin to even have words in them

**Comment by Rperry2174** [comment]
Yeah that might work and I don't think we would add the github logo to every box, just the one in focus (when a box is in focus). So in the picture above it would only be that row that has the logo inside of it.

The reason for this is because what you said about the boxes that are too thin.. If you have selected a particular node (aka clicked on one), then the nodes above it dim and the node itself becomes the "base" and takes up 100% of the width.

I think this would ensure that most of the time when the logo is included in this "focused" node then it wont get too crowded

**Comment by Rperry2174** [comment]
Just made our demo repos public so this could work now:
Python: https://github.com/pyroscope-io/hotrod-python
Ruby: https://github.com/pyroscope-io/hotrod-ruby
Go: https://github.com/pyroscope-io/jaeger.git

/cc @LouisInFlow

**Comment by aleks-p** [comment]
Closing this since [source code integration](https://grafana.com/docs/pyroscope/latest/view-and-analyze-profile-data/line-by-line/) has been available for a while. Pyroscope provides the API components, the UI is available in Grafana via Drilldown Profiles. 

There are no plans to bring alternative UIs at this point.
